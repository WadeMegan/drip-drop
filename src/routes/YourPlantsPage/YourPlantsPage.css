.yourPlantsContainer { 
    width:100%;
    padding-bottom:20px;
    padding-left:20px;
    padding-right:20px;
    display:flex;
    flex-direction: column;
    justify-content:center;
    align-items:center;
}

.directions {
    color:grey;
}

.directions a {
    color: black;
}

.directions a:hover {
    color: grey;
    cursor:pointer;
}

#searchPlants {
    border:1px solid black;
}

.searchButton {
    border: 1px solid black;
    background-color: grey;
    color:white;
}

.searchButton:hover {
    background-color: black;
    color:white;    
    cursor: pointer;
}

.usersPlantsSection {
    padding-top:6vh;
}