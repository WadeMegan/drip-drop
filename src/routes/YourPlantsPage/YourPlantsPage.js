import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import './YourPlantsPage.css'
import PlantListContext from '../../contexts/PlantListContext'
import PlantItem from '../../components/PlantItem/PlantItem'
import PlantApiService from '../../services/plant-api-services'
import UserService from '../../services/user-service'
import Error from '../../components/Error/Error'

export default class YourPlantsPage extends Component {
    static contextType = PlantListContext

    state = {
        isFiltered:false,
        filteredPlants: []
    }

    componentDidMount(){
        this.context.clearError()
        const userId = UserService.getUserToken()

        if(!this.context.usersPlants.length){
            PlantApiService.getUsersPlants(userId)
                .then(this.context.setUsersPlants)
                .catch(this.context.setError)
        }
        
        if(!this.context.plantList.length){
            PlantApiService.getAllPlants()
                .then(this.context.setPlantList)
                .catch(this.context.setError)
        }
    }

    componentWillUnmount(){
        this.context.clearError()
    }

    renderPlants=(plants)=>{
        return plants.map(plant =>
            <PlantItem key={plant.id} plant={plant}/>
        )
    }

    filterPlants = (event) => {
        let clonedArray = JSON.parse(JSON.stringify(this.context.usersPlants))

        let plants = clonedArray.filter((plant)=>{
            return plant.name.toLowerCase().search(event.target.value.toLowerCase()) !== -1
        })

        this.setState({
            isFiltered:true,
            filteredPlants: plants
        })
    }

    render(){
        return (
            <Error>
                <div className='switchPagesNav'>
                    <Link className='pageButton directionsButton notCurrent' to='directions'>
                        <p className='buttonItem'>Directions</p>
                    </Link>
                    <Link className='pageButton availableButton notCurrent' to='plants'>
                        <p className='buttonItem'>Available Plants</p>
                    </Link>
                    <div className='pageButton yourPlantsButton current'>
                        Your Plants
                    </div>
                </div>
                <section className='usersPlantsSection'>
                    <div className='banner'>
                        <h2>Your Plants</h2>
                        <form className='searchForm'> 
                            <label htmlFor='searchPlants'>Search your plants: </label>
                            <input type='text' name='searchPlants' id='searchPlants' placeholder='zz plant' onChange={this.filterPlants}/>
                        </form>
                    </div>
                    <div className='yourPlantsContainer'>
                        {this.state.isFiltered
                        ? this.renderPlants(this.state.filteredPlants)
                        : this.renderPlants(this.context.usersPlants)}
                    </div>
                </section>
            </Error>
        )
    }
}