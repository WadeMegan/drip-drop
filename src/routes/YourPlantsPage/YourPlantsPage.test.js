import React from 'react'
import ReactDOM from 'react-dom'
import YourPlantsPage from './YourPlantsPage'
import {BrowserRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'

it('renders without crashing',()=>{
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><YourPlantsPage/></BrowserRouter>,div)
    ReactDOM.unmountComponentAtNode(div)
})

it('should match with snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <YourPlantsPage/>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot()
})
