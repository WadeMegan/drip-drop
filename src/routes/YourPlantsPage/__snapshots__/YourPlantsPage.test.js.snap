// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
Array [
  <div
    className="switchPagesNav"
  >
    <a
      className="pageButton directionsButton notCurrent"
      href="/directions"
      onClick={[Function]}
    >
      <p
        className="buttonItem"
      >
        Directions
      </p>
    </a>
    <a
      className="pageButton availableButton notCurrent"
      href="/plants"
      onClick={[Function]}
    >
      <p
        className="buttonItem"
      >
        Available Plants
      </p>
    </a>
    <div
      className="pageButton yourPlantsButton current"
    >
      Your Plants
    </div>
  </div>,
  <section
    className="usersPlantsSection"
  >
    <div
      className="banner"
    >
      <h2>
        Your Plants
      </h2>
      <form
        className="searchForm"
      >
        <label
          htmlFor="searchPlants"
        >
          Search your plants: 
        </label>
        <input
          id="searchPlants"
          name="searchPlants"
          onChange={[Function]}
          placeholder="zz plant"
          type="text"
        />
      </form>
    </div>
    <div
      className="yourPlantsContainer"
    />
  </section>,
]
`;
